libraries/AP_Notify/DiscoLED.h

```diff
@@ -0,0 +1,39 @@
+/*
+  Copyright (C) 2016 Mathieu Othacehe. All rights reserved.
+
+  This program is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  This program is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+#pragma once
+
+#include <AP_HAL/AP_HAL.h>
+#if CONFIG_HAL_BOARD == HAL_BOARD_LINUX
+#include <AP_HAL_Linux/PWM_Sysfs.h>
+#include "RGBLed.h"
+
+class DiscoLED: public RGBLed
+{
+public:
+    DiscoLED();
+    bool hw_init(void);
+    bool hw_set_rgb(uint8_t r, uint8_t g, uint8_t b);
+private:
+    Linux::PWM_Sysfs_Bebop *red_pwm;
+    Linux::PWM_Sysfs_Bebop *green_pwm;
+    Linux::PWM_Sysfs_Bebop *blue_pwm;
```

Comment:
ahh, ok.  It's because of the way we add objects to a global "hal" object in HAL_Linux_Class.cpp. The constructor of each object is called in the order they appear in that file and then we construct a AP_HAL_Linux with the instance of each other object.

So we can't always call `hal.xxxxx()` inside constructors and usually postpone the call to be done inside an `init()` method. I fixed in some places reordering the init or passing the instance so we don't rely on the `hal` object.  I don't think it's worth in this case. I'm fine with the extra allocation.

It's possible to fix